relay-server/src/services/processor.rs

```diff
@@ -1274,23 +1274,41 @@ impl EnvelopeProcessorService {
         event::finalize(state, &self.inner.config)?;
         self.light_normalize_event(state)?;
         dynamic_sampling::normalize(state);
-        event::filter(state)?;
-        dynamic_sampling::run(state, &self.inner.config);
+        event::filter(state, &self.inner.global_config.current())?;
+        // Don't extract metrics if relay can't apply generic inbound filters.
+        // An inbound filter applied in another up-to-date relay in chain may
+        // need to drop the event, and there should not be metrics from dropped
+        // events.
+        // In processing relays, always extract metrics to avoid losing them.
+        let supported_generic_filters = self.inner.global_config.current().filters.is_ok()
+            && relay_filter::are_generic_filters_supported(
+                self.inner
+                    .global_config
+                    .current()
+                    .filters()
+                    .map(|f| f.version),
+                state.project_state.config.filter_settings.generic.version,
+            );
 
-        // Remember sampling decision, before it is reset in `dynamic_sampling::sample_envelope_items`.
-        let sampling_should_drop = state.sampling_result.should_drop();
+        let mut sampling_should_drop = false;
 
-        // We avoid extracting metrics if we are not sampling the event while in non-processing
-        // relays, in order to synchronize rate limits on indexed and processed transactions.
-        if self.inner.config.processing_enabled() || sampling_should_drop {
-            self.extract_metrics(state)?;
-        }
+        if self.inner.config.processing_enabled() || supported_generic_filters {
```

Comment:
Shouldn't this be:

```suggestion
        if self.inner.config.processing_enabled() && supported_generic_filters {
```

Reply:
Nope: processing relays should always run dynamic sampling, even if generic filters are unsupported.

Reply:
Managed Relays also have processing enabled!

Reply:
They shouldn't, but we've seen customers do it. IMO `Processing` should be a mode, just like `Managed`.